Apromore-ProcessMining-Collection/prom-bpmn/src/main/java/org/apromore/processmining/models/graphbased/directed/bpmn/ModelCheckResult.java

```diff
@@ -0,0 +1,102 @@
+    public static final int END_NO_INCOMING_ARC_CODE = 13;
+    public static final int END_OUTGOING_ARCS_CODE = 14;
+
+    private static final String DISCONNECTED_ARC_MSG = "The model has disconnected arcs";
+    private static final String EMPTY_MODEL_MSG = "The model is empty";
+    private static final String POOLS_NOT_SUPPORTED_MSG = "There are pools in the model. Pools are not yet supported";
+    private static final String SELF_LOOP_MSG_FORMAT = "The element %s has a self-loop";
+    private static final String TASK_MULTIPLE_OUTGOING_ARCS_MSG_FORMAT = "The task %s has more than one outgoing arc";
+    private static final String TASK_MULTIPLE_INCOMING_ARCS_MSG_FORMAT = "The task %s has more than one incoming arc";
+    private static final String TASK_MISSING_ARCS_MSG_FORMAT = "The task %s has missing incoming or outgoing arcs";
+    private static final String GATE_MISSING_ARCS_MSG_FORMAT = "The gateway %s has missing incoming or outgoing arcs";
+    private static final String START_MULTIPLE_OUTGOING_ARCS_MSG = "The Start Event has more than one outgoing arc";
+    private static final String START_NO_OUTGOING_ARC_MSG = "The Start Event has a missing outgoing arc";
+    private static final String START_INCOMING_ARCS_MSG = "The Start Event has incoming arc(s)";
+    private static final String END_MULTIPLE_INCOMING_ARCS_MSG = "The End Event has more than one incoming arc";
+    private static final String END_NO_INCOMING_ARC_MSG = "The End Event has a missing incoming arc";
+    private static final String END_OUTGOING_ARCS_MSG = "The End Event has outgoing arc(s)";
+
+    private static final Map<Integer, String> ERROR_MAP = Map.ofEntries(
+            Map.entry(DISCONNECTED_ARC_CODE, DISCONNECTED_ARC_MSG),
+            Map.entry(EMPTY_MODEL_CODE, EMPTY_MODEL_MSG),
+            Map.entry(POOLS_NOT_SUPPORTED_CODE, POOLS_NOT_SUPPORTED_MSG),
+            Map.entry(SELF_LOOP_CODE, SELF_LOOP_MSG_FORMAT),
+            Map.entry(TASK_MULTIPLE_OUTGOING_ARCS_CODE, TASK_MULTIPLE_OUTGOING_ARCS_MSG_FORMAT),
+            Map.entry(TASK_MULTIPLE_INCOMING_ARCS_CODE, TASK_MULTIPLE_INCOMING_ARCS_MSG_FORMAT),
+            Map.entry(TASK_MISSING_ARCS_CODE, TASK_MISSING_ARCS_MSG_FORMAT),
+            Map.entry(GATE_MISSING_ARCS_CODE, GATE_MISSING_ARCS_MSG_FORMAT),
+            Map.entry(START_MULTIPLE_OUTGOING_ARCS_CODE, START_MULTIPLE_OUTGOING_ARCS_MSG),
+            Map.entry(START_NO_OUTGOING_ARC_CODE, START_NO_OUTGOING_ARC_MSG),
+            Map.entry(START_INCOMING_ARCS_CODE, START_INCOMING_ARCS_MSG),
+            Map.entry(END_MULTIPLE_INCOMING_ARCS_CODE, END_MULTIPLE_INCOMING_ARCS_MSG),
+            Map.entry(END_NO_INCOMING_ARC_CODE, END_NO_INCOMING_ARC_MSG),
+            Map.entry(END_OUTGOING_ARCS_CODE, END_OUTGOING_ARCS_MSG)
+    );
+
+    private Map<Integer, List<String>> errors;
+
+    public ModelCheckResult(Map<Integer, List<String>> errorMap) {
+        this.errors = errorMap;
+    }
```

Comment:
To be safe, should put a @NonNull constraint for input param errorMap; otherwise, the other code using errorMap will throw NPE.